ibft/event_loop.go

```diff
@@ -0,0 +1,96 @@
+package ibft
+
+import (
+	"github.com/bloxapp/ssv/network/msgqueue"
+	"github.com/prysmaticlabs/prysm/shared/mathutil"
+	"go.uber.org/zap"
+	"time"
+)
+
+/**
+Multi threading in iBFT instance -
+
+The iBFT instance is a single thread service.
+All the different events (reading network messages, timeouts, etc) are brokered through an event queue.
+
+Events are added asynchronically to the queue but are pulled synchronically in just one place, the main event loop.
+
+
+
+***** async ******\\\\\\\\\\\\\\\\\\  sync  \\\\\\\\\\\\\\\\\\\\\\\
+Network message->|
+Timeouts->		 |-> added to event queue -> pulled from event queue
+Other events->   |
+
+
+*/
+
+// StartMainEventLoop start the main event loop queue for the iBFT instance which iterates events in the queue, if non found it will wait before trying again.
+func (i *Instance) StartMainEventLoop() {
+	for {
+		if i.Stopped() {
+			return
+		}
+
+		if f := i.eventQueue.Pop(); f != nil {
+			f()
+		} else {
+			time.Sleep(time.Millisecond * 100)
+		}
+	}
+}
+
+// StartMessagePipeline - the iBFT instance is message driven with an 'upon' logic.
+// each message type has it's own pipeline of checks and actions, called by the networker implementation.
+// Internal chan monitor if the instance reached decision or if a round change is required.
+func (i *Instance) StartMessagePipeline() {
+	for {
+		if i.Stopped() {
+			return
+		}
+
+		// TODO - refactor
+		if i.MsgQueue.MsgCount(msgqueue.IBFTRoundIndexKey(i.State.Lambda, i.State.SeqNumber, i.State.Round)) > 0 {
```

Comment:
Need to change the of the key index name to "IBFTMessageIndexKey" or so

Reply:
@nivBlox ok